src/Microsoft.AspNet.Mvc.Core/ActionResults/PartialViewResult.cs

```diff
@@ -50,10 +51,22 @@ public class PartialViewResult : ActionResult
             var viewEngine = ViewEngine ??
                              context.HttpContext.RequestServices.GetRequiredService<ICompositeViewEngine>();
 
+            var logger = context.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>()
+                .CreateLogger<PartialViewResult>();
+
             var viewName = ViewName ?? context.ActionDescriptor.Name;
-            var view = viewEngine.FindPartialView(context, viewName)
-                                 .EnsureSuccessful()
-                                 .View;
+            var viewEngineResult = viewEngine.FindPartialView(context, viewName);
+            if (!viewEngineResult.Success)
+            {
+                logger.LogError(
+                    "The partial view '{PartialViewName}' was not found. Searched locations: {SearchedViewLocations}",
+                    viewName,
+                    viewEngineResult.SearchedLocations);
+            }
+
+            var view = viewEngineResult.EnsureSuccessful().View;
+
+            logger.LogVerbose("The partial view '{PartialViewName}' was found successfully.", viewName);
```

Comment:
@Eilon views to me seem like one of those things that's first class enough to be worth it. Especially because there's a search algorithm behind it that's not well understood by many users.